Update library to work with react v0.14

everdimension opened the pull request:
I rewrote the code for this library to remove deprecated stuff:
- `ReactDOM.render` instead of `React.render`
- Removed `jsx` pragma as it is no longer unnecessary
- `React.cloneElement()` insteat of `cloneWithProps()`
- The react `props` are now "read only" and cannot be mutated.
- Used `.jsx` extention for the MagicMove code and added `npm prepublish` script to turn `.jsx` to `.js` for correct usage as a node_module.
- Added a script for bundling example files

The library now works with latest react.

everdimension committed:
update package building setup

everdimension committed:
update react; rewrite MagicMove to work with react v0.14 and react-dom

everdimension committed:
update code for examples

ryanflorence commented:
if you could remove the `.jsx` extensions and remove the built files, I can merge this. Thanks :)

everdimension committed:
rm built files

everdimension committed:
restructure source files, build as a module

everdimension commented:
Ok, I have removed the generated files for examples as well as for the library itself.
I also added a `src` folder to make source files more clear.

`npm run build` or `npm run prepublish` will build appropriately.

balupton commented:
Is this fork published to npm somewhere?

everdimension committed:
autogenerate lib files when installed from git

everdimension committed:
upd dependencies

everdimension committed:
fix package.json syntax

everdimension committed:
update postinstall script

everdimension committed:
rm unnecessary dependencies

everdimension committed:
rm unnecessary dependencies, update build scripts

everdimension committed:
add built lib files to gitignore; move dev modules to devDependencies

everdimension commented:
@balupton 
I have updated `npm scripts` for this pull request (and cleaned up dependencies) to `postinstall` everything needed when being installed from git, so until this PR is merged you can install this library as a dependency directly from the git branch, like this:
```bash
npm install --save git://github.com/everdimension/react-magic-move.git#update_code
```

everdimension commented:
@ryanflorence if you have time to merge this one it'd be great

cmseaton42 commented:
Was this branch ever merged?

everdimension commented:
@cmseaton42 No, not yet.

@ryanflorence did say that this project is not likely to be supported.

cmseaton42 commented:
@everdimension true dat. 

I found that the 'react-flip-move' package is a good alternative.
